### DISCUSSION TITLE
Automation and fast when selling

### DISCUSSION THREAD

**Discussion by osvaldoclasico** [Blair2004/NexoPOS#616] [Ideas]
I've been checking some of the functions of the pos system, so far so good, but I have some ideas that you can consider for your next update.
With these ideas I base myself on the speed that you have to have when selling, currently to change the quantities it takes a little time to select and then write the quantities. With these points I think the system would be more favorable and flexible when it comes to some quick sales.

1- Symbolizes the remainder of the current quantity in pos.

Example:

Client: I need 970 units of Product (X).

Cashier: Write the required amount: 970.

Customer: I'm hesitant, give me 969, it's easier to click on the (-) than to write the amount again.

2- Symbolizes the input of the quantities requested, it can be left like this so that when you click you can modify the quantity to your liking.

3- Symbolizes the sum of a new amount, let's go back to the previous exercise

Customer: I'm hesitant, give me 971, it's easier to click on the (+) than to write the amount again.

4- symbolizes that you only have X Quantity of products.

Example:
If of Product X the current quantity = 970 Stock.

Client: I need 971 units of product X but in this case we have 970, it would be better if when entering the amount 971, only 970 is added automatically in the input (2) and the alert is thrown (4).

With this, it would be faster if the client decides and says, well, give me the current amounts, in this case you will no longer have to write and the process would be a little more automated.

Taking into account all these points we can use the keyboard shortcuts to add the + and - of the same

here some images
https://ibb.co/8xz9Nhv
https://ibb.co/4MF1qdm

**Comment by Blair2004** [comment]
Hi, 
Thank you for your contribution. 
- I agree with the fact we need a way to adjust quantities more quickly. Here is a sample i've prepared:

![screenshot-nimbusweb me-2022 03 26-06_29_19](https://user-images.githubusercontent.com/5265663/160226245-476ec41f-68c3-48f2-9407-ebfef7f4356a.jpg)

- adding the current available quantity even if it's less than what the customer requested... i don't really agree with that, in case the customer want exactly that quantity, the cashier will be forced to remove the product from the cart. I rather believe we should have the available quantity visible for the cashier.

![screenshot-nimbusweb me-2022 03 26-06_33_39](https://user-images.githubusercontent.com/5265663/160226325-c3c866d9-a6e3-4113-85cb-c3dea4727b50.jpg)

**Reply by osvaldoclasico** [reply]
> Hola, gracias por tu aporte.
> 
> * Estoy de acuerdo con el hecho de que necesitamos una forma de ajustar las cantidades más rápidamente. Aquí hay una muestra que he preparado:
> 
> ![captura de pantalla-nimbusweb me-2022 03 26-06_29_19](https://user-images.githubusercontent.com/5265663/160226245-476ec41f-68c3-48f2-9407-ebfef7f4356a.jpg)
> 
> * agregando la cantidad disponible actual incluso si es menor a lo que el cliente solicitó... realmente no estoy de acuerdo con eso, en caso de que el cliente quiera exactamente esa cantidad, el cajero se verá obligado a retirar el producto del carrito. Más bien creo que deberíamos tener la cantidad disponible visible para el cajero.
> 
> ![captura de pantalla-nimbusweb me-2022 03 26-06_33_39](https://user-images.githubusercontent.com/5265663/160226325-c3c866d9-a6e3-4113-85cb-c3dea4727b50.jpg)

Simply magnificent, I agree with your idea, it would look better on top 

As for the quantities totally agree !!
